tests/integration_tests/performance/test_vsock_throughput.py

```diff
@@ -0,0 +1,381 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Tests the VSOCK throughput of Firecracker uVMs."""
+
+
+import os
+import json
+import logging
+import time
+import concurrent.futures
+import pytest
+from conftest import _test_images_s3_bucket
+from framework.artifacts import ArtifactCollection, ArtifactSet
+from framework.matrix import TestMatrix, TestContext
+from framework.builder import MicrovmBuilder
+from framework.statistics import core, consumer, producer, criteria,  types, \
+    function
+from framework.utils import CpuMap, CmdBuilder, run_cmd, eager_map, \
+    get_cpu_percent, get_cpu_model_name
+import host_tools.network as net_tools
+from integration_tests.performance.vsock_throughput_test_config import \
+    THROUGHPUT, DURATION, CPU_UTILIZATION_HOST, \
+    CPU_UTILIZATION_GUEST, CONFIG, CPU_UTILIZATION_VMM_TAG, \
+    CPU_UTILIZATION_VCPUS_TOTAL_TAG, THROUGHPUT_TOTAL, \
+    DURATION_TOTAL, IPERF3, BASE_PORT, \
+    IPERF3_CPU_UTILIZATION_PERCENT_OUT_TAG, IPERF3_END_RESULTS_TAG, \
+    TARGET_TAG, DELTA_PERCENTAGE_TAG, THROUGHPUT_UNIT, DURATION_UNIT, \
+    CPU_UTILIZATION_UNIT
+
+SERVER_STARTUP_TIME = CONFIG["server_startup_time"]
+
+VSOCK_UDS_PATH = "v.sock"
+
+
+def measurements_vsock():
+    """Define the produced measurements for VSOCK workloads."""
+    return [types.MeasurementDef(THROUGHPUT, THROUGHPUT_UNIT),
```

Comment:
No latency ?

Reply:
no latency for this test, iperf3 does not support it: https://github.com/esnet/iperf/issues/547
We could add support for this later on, if we see the need, using other tools. But this would be a separate test

IMHO, i expect that any regression in latency would also be caught by a throughput test

Reply:
Latency is also a metric that we care about and I would expect the latency to impact tput only when we are maxing out on CPU usage. I would want to measure latency during lightweight/moderate/stress traffi.  If this scenario is not supported by the current tool, please open a ticket so we pick this up in the future.

Reply:
Opened an issue for VSOCK latency perf test: https://github.com/firecracker-microvm/firecracker/issues/2353